### ISSUE TITLE
Stuck forever when formatting erb file

### ISSUE THREAD

**Issue by EnziinSystem** [threedaymonk/htmlbeautifier#58]
Hello,

I install ERB Formatter/Beautify plugin in VS Code and format the attachment file, but it stuck forever with heavy CPU and overheat.

If I running the command from terminal:
```
htmlbeautifier app/views/...file_name.html.eb
```
It stuck forever in terminal command.

```
kevin@local:~/developer/$ htmlbeautifier app/views/common/file_name.html.erb
^CTraceback (most recent call last):
	18: from /home/kevin/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `<main>'
	17: from /home/kevin/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `eval'
	16: from /home/kevin/.rvm/gems/ruby-2.7.2/bin/htmlbeautifier:23:in `<main>'
	15: from /home/kevin/.rvm/gems/ruby-2.7.2/bin/htmlbeautifier:23:in `load'
	14: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/bin/htmlbeautifier:70:in `<top (required)>'
	13: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/bin/htmlbeautifier:70:in `each'
	12: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/bin/htmlbeautifier:73:in `block in <top (required)>'
	11: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/bin/htmlbeautifier:73:in `open'
	10: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/bin/htmlbeautifier:74:in `block (2 levels) in <top (required)>'
	 9: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/bin/htmlbeautifier:7:in `beautify'
	 8: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier.rb:25:in `beautify'
	 7: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier.rb:25:in `tap'
	 6: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier.rb:26:in `block in beautify'
	 5: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier/parser.rb:16:in `scan'
	 4: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier/parser.rb:30:in `dispatch'
	 3: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier/parser.rb:30:in `find'
	 2: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier/parser.rb:30:in `each'
	 1: from /home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier/parser.rb:30:in `block in dispatch'
/home/kevin/.rvm/gems/ruby-2.7.2/gems/htmlbeautifier-1.3.1/lib/htmlbeautifier/parser.rb:30:in `scan': Interrupt

```

Please use the attached file for a test.

[show.html.zip](https://github.com/threedaymonk/htmlbeautifier/files/7517348/show.html.zip)

**Comment by jeremyburks-sikich** [comment]
I've observed the same behavior. In my case, `case` statements in erb templates seem to cause the problem.

**Comment by threedaymonk** [comment]
This should be fixed in version 1.4.0. Please check and let me know if not.

**Comment by jeremyburks-sikich** [comment]
Thanks for the release and your response! Version 1.4.0 offers an improvement. `htmlbeautifier` no longer hangs. Instead, I receive the error `Extraneous closing tag on line 168 (RuntimeError)`. I've confirmed that `htmlbeautifier` processes the template without error with the `case` statements removed. 

It looks like `HtmlBeautifier::Builder`  might need some special handling for `case` statements. I could be off on that since I don't fully grok the library yet.
